### PR TITLE
Add blobber conductor tests

### PR DESCRIPTION
**Pull request by fabioDMFerreira**
#1651 

* Add blobber test that verifies whether a directory listing result is correct even if one malicious blobber sends wrong data
* Add blobber test that verifies whether a directory listing result is correct even if one malicious blobber sends wrong metadata
* Add blobber test that verifies whether listing/deleting/downloading/uploading is successful even if one malicious blobber does not respond
* Add blobber test that verifies whether listing/deleting/downloading/uploading is successful even if one malicious blobber returns an error

* Add validator test that verifies whether a challenge is validated even if there is an adversarial validator failing on the challenge validation.
* Add validator test that verifies whether a challenge is invalid even if there is an adversarial validator validating the challenge.
* Add validator test that verifies whether a challenge is validated even if there is an adversarial validator not responding with the validation result.

### CODE REVIEW

File: code/go/0chain.net/chaincore/transaction/entity.go [hunk 1]
```diff
@@ -415,7 +413,7 @@ DebugTxn - is this a transaction that needs being debugged
 - applicable only when running in test mode and the transaction_data string contains debug keyword somewhere in it
 */
 func (t *Transaction) DebugTxn() bool {
-	return config.Development() && viper.GetBool("logging.verbose")
+	return true
```
Comment (dabasov) [inline code/go/0chain.net/chaincore/transaction/entity.go]: we have config option for this now, revert this change to avoid merge conflicts

File: code/go/0chain.net/conductor/conductrpc/server.go [hunk 2]
```diff
@@ -397,10 +397,30 @@ func (s *Server) State(id NodeID, state *State) (err error) {
...
+	// so we can return	an existing state of a different node.
+	if strings.Contains(string(id), "validator-") {
+		for _, k := range s.names {
+			if strings.Contains(string(k), "blobber-") {
+				name = k
+				break
+			}
+		}
+		var ns *nodeState
+		if ns = s.nodes[name]; err != nil {
```
Comment (dabasov) [inline code/go/0chain.net/conductor/conductrpc/server.go]: err != nil is always false

File: code/go/0chain.net/miner/protocol_block_integration_tests.go [hunk 3]
```diff
@@ -78,18 +81,64 @@ func hasDST(pb, b []*transaction.Transaction) (has bool) {
...
-func (mc *Chain) isTestingOnUpdateFinalizedBlock(round int64) bool {
+func addResultIfAdversarialValidatorTest(b *block.Block) {
+	s := crpc.Client().State()
+
+	if s.AdversarialValidator == nil || !s.IsMonitor {
+		return
+	}
+
+	for _, tx := range b.Txns {
+		var txParsed map[string]interface{}
```
Comment (dabasov) [inline code/go/0chain.net/miner/protocol_block_integration_tests.go]: do we need txParsed?

File: code/go/0chain.net/smartcontract/storagesc/blobber_integration_tests.go [hunk 4]
```diff
@@ -22,3 +24,30 @@ func afterInsertBlobber(id string) {
...
+func beforeEmitAddChallenge(challenge *StorageChallengeResponse) {
+	var (
+		client = crpc.Client()
+		state  = client.State()
+	)
+
+	if state.AdversarialValidator != nil && state.AdversarialValidator.PassAllChallenges && containsString(challenge.ValidatorIDs, state.AdversarialValidator.ID) {
+		// any challenge adulteration produces an invalid challenge
+		challenge.AllocationRoot = strings.ReplaceAll(challenge.AllocationRoot, "1", "0")
+		challenge.AllocationRoot = strings.ReplaceAll(challenge.AllocationRoot, "a", "b")
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/blobber_integration_tests.go:43]: if we want to break Allocation root we can shuffle chars probably, this one isnot guaranteed to break it

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/chaincore/transaction/entity.go (hunk 1 above)
Comment (fabioDMFerreira) [reply code/go/0chain.net/chaincore/transaction/entity.go]: fixed by [8843e24](https://github.com/0chain/0chain/pull/1603/commits/8843e249a2cc8643e5309411e150b118bfd53f4e)

File: code/go/0chain.net/conductor/conductrpc/server.go (hunk 2 above)
Comment (fabioDMFerreira) [reply code/go/0chain.net/conductor/conductrpc/server.go]: fixed by [8843e24](https://github.com/0chain/0chain/pull/1603/commits/8843e249a2cc8643e5309411e150b118bfd53f4e)

File: code/go/0chain.net/miner/protocol_block_integration_tests.go (hunk 3 above)
Comment (fabioDMFerreira) [reply code/go/0chain.net/miner/protocol_block_integration_tests.go]: fixed by [8843e24](https://github.com/0chain/0chain/pull/1603/commits/8843e249a2cc8643e5309411e150b118bfd53f4e)

File: code/go/0chain.net/smartcontract/storagesc/blobber_integration_tests.go (hunk 4 above)
Comment (fabioDMFerreira) [reply code/go/0chain.net/smartcontract/storagesc/blobber_integration_tests.go:43]: Although it is very unlikely to have the string without an `a` and a `1` it may happen. Changing the implementation to revert the string.

Comment (fabioDMFerreira) [reply code/go/0chain.net/smartcontract/storagesc/blobber_integration_tests.go:43]: fixed by [1ad7afe](https://github.com/0chain/0chain/pull/1603/commits/1ad7afeae1c4d31e6873fca49d02796187ba507d)

**Review by dabasov — APPROVED** [review]
lgtm
